includes/abstracts/llms-abstract-admin-wizard.php

```diff
@@ -0,0 +1,390 @@
+
+	/**
+	 * Steps
+	 *
+	 * @since [version]
+	 *
+	 * @var array
+	 */
+	protected array $steps;
+
+	/**
+	 * Page title.
+	 *
+	 * @since [version]
+	 *
+	 * @var string
+	 */
+	protected string $title;
+
+	/**
+	 * Error message.
+	 *
+	 * @since [version]
+	 * @since [version]
+	 *
+	 * @var WP_Error|null
+	 */
+	protected ?WP_Error $error = null;
+
+	/**
+	 * Optional transient key to store wizard data.
+	 *
+	 * @since [version]
+	 *
+	 * @var string
+	 */
+	protected string $transient_key = '';
+
+	/**
+	 * Add hooks.
+	 *
+	 * @since [version]
+	 *
+	 * @return void
+	 */
+	protected function add_hooks(): void {
+
+		/**
+		 * Whether the LifterLMS Wizard is enabled.
+		 *
+		 * This filter may be used to entirely disable the setup wizard.
+		 *
+		 * @since [version]
+		 *
+		 * @param boolean $enabled Whether the wizard is enabled.
+		 */
+		if ( apply_filters( "llms_enable_{$this->type}_wizard", true ) ) {
+			add_action( 'admin_enqueue_scripts', array( $this, 'enqueue' ) );
+			add_action( 'admin_menu', array( $this, 'admin_menu' ) );
+			add_action( 'admin_init', array( $this, 'save' ) );
+		}
+	}
+
+	/**
+	 * Register wizard setup page
+	 *
+	 * @since [version]
+	 *
+	 * @return string The hook suffix of the setup wizard page ("admin_page_llms-setup"), or `false` if the user does not have the capability required.
+	 */
+	public function admin_menu(): string {
+
+		/**
+		 * Filter the WP User capability required to access and run the setup wizard.
+		 *
+		 * @since [version]
+		 *
+		 * @param string $cap Required user capability. Default value is `install_plugins`.
+		 */
+		$cap = apply_filters( 'llms_setup_wizard_access', 'install_plugins' );
+
+		$hook = add_dashboard_page(
+			$this->title,
+			'',
+			$cap,
+			'llms-' . $this->type,
+			array( $this, 'output' )
+		);
+
+		update_option( 'lifterlms_first_time_' . $this->type, 'yes' );
+
+		return $hook;
+	}
+
+	/**
+	 * Enqueue static assets for the setup wizard screens
+	 *
+	 * @since [version]
+	 *
+	 * @return bool
+	 */
+	public function enqueue(): bool {
+
+		if ( ! isset( $_GET['page'] ) || 'llms-' . $this->type !== $_GET['page'] ) {
+			return '';
+		}
+
+		$extra = true;
+
+		if ( 'finish' === $this->get_current_step() ) {
+			$extra = llms()->assets->enqueue_style( 'llms-admin-importer' );
+		}
+
+		return llms()->assets->enqueue_script( 'llms-admin-setup' ) && llms()->assets->enqueue_style( 'llms-admin-setup' ) && $extra;
+
+	}
+
+	/**
+	 * Retrieve the redirect URL to use after an import is complete at the conclusion of the wizard
+	 *
+	 * If a single course is imported, redirects to that course's edit page, otherwise redirects
+	 * to the course post table list sorted by created date with the most recent courses first.
+	 *
+	 * @since [version]
+	 *
+	 * @param int[] $course_ids WP_Post IDs of the course(s) generated during the import.
+	 * @return string
+	 */
+	protected function get_completed_url( array $course_ids ): string {
```

Comment:
this implementation is something to be kept in the **setup** wizard only.
By default it can be something all the exteinding class must implement:
`	protected function get_completed_url( array $course_ids ): string;`